### PR TITLE
add Sonatype-based publishing

### PR DESCRIPTION
**Pull request by SethTisue**
modeled after playframework/play-json#603

### CODE REVIEW

**Comment by SethTisue** [comment]
@octonato can you add the same publishing secrets to Travis-CI that you added in the play-json repo?

File: project/plugins.sbt [hunk 1]
```diff
@@ -2,4 +2,4 @@ addSbtPlugin("com.typesafe"      % "sbt-mima-plugin"       % "0.9.2")
 addSbtPlugin("org.scalastyle"   %% "scalastyle-sbt-plugin" % "1.0.0")
 addSbtPlugin("de.heikoseeberger" % "sbt-header"            % "5.6.0")
 addSbtPlugin("org.scalameta"     % "sbt-scalafmt"          % "2.4.2")
-addSbtPlugin("com.dwijnand"      % "sbt-dynver"            % "4.1.1")
```
Comment (octonato) [inline project/plugins.sbt:5]: Why remove dynver? Is that now included by `sbt-ci-release`?

File: build.sbt [hunk 2]
```diff
@@ -27,6 +27,7 @@ lazy val cachecontrol = (project in file("."))
       organization.value %% name.value % previousStableVersion.value
         .getOrElse(throw new Error("Unable to determine previous version"))
     ),
+    sonatypeProfileName := "com.typesafe",
```
Comment (octonato) [inline build.sbt:30]: Great, we can use our new skills! 😄

**Review by octonato — APPROVED** [review]
LGTM

**Comment by octonato** [comment]
Secrets added

File: project/plugins.sbt (hunk 1 above)
Comment (SethTisue) [reply project/plugins.sbt:5]: exactly

**Comment by SethTisue** [comment]
let's try it out at #182

**Pull request merged**